latex/generator-tex.sh

```
@@ -0,0 +1,133 @@
+    : > "$MERMAID_FILE"  # Clear file
+    while IFS= read -r mermaid_line; do
+      if [[ "$mermaid_line" == '```' ]]; then
+        break
+      fi
+      echo "$mermaid_line" >> "$MERMAID_FILE"
+    done
+
+    # Render to high-resolution PNG
+    echo "Rendering mermaid block $MERMAID_COUNTER to high-res PNG..."
+    if ! mmdc -i "$MERMAID_FILE" -o "$PNG_FILE" 
+         --outputFormat png \
```

Shellcheck (0.11.0)

[warning] This flag is used as a command name. Bad line break or missing [ .. ]?

(SC2215)